discovery-provider/tests/test_ipfs_functionality.py

```diff
@@ -40,3 +40,50 @@ def test_ipfs(app):
     for key in test_metadata_object:
         assert key in ipfs_metadata
         assert test_metadata_object[key] == ipfs_metadata[key]
+
+# test for ipfs_lib.py, multihash_is_directory
+class TestMultihashIsDirectory:
+    # Invoke multihash_is_directory with a invalid cid, verify that it throws an 'invalid multihash' error
+    def test_invalid_cid(self, app):
+        ipfs_peer_host = app.config["ipfs"]["host"]
+        ipfs_peer_port = app.config["ipfs"]["port"]
+
+        # Instantiate IPFS client from src lib
+        ipfsclient = IPFSClient(ipfs_peer_host, ipfs_peer_port, [])
+
+        try:
+            ipfsclient.multihash_is_directory('Qmfake')
+        except Exception as e:
+            assert 'invalid multihash' in str(e)
+
+    # Invoke multihash_is_directory with cat function handler that returns some other response
+    def test_valid_cid_directory(self, app):
+        ipfs_peer_host = app.config["ipfs"]["host"]
+        ipfs_peer_port = app.config["ipfs"]["port"]
+
+        # Instantiate IPFS client from src lib
+        ipfsclient = IPFSClient(ipfs_peer_host, ipfs_peer_port, [])
+
+        def cat_handler_dir(cid, s, e, timeout):
```

Comment:
this cat handler pattern is cool, can we document it a little bit inline

Reply:
yep will do

Reply:
added a comment here. let me know what you think